charts/covidDataExplorer/CovidDataExplorer.tsx

```diff
@@ -751,22 +753,47 @@ export class CovidDataExplorer extends React.Component<{
         }
     }
 
-    private _updateMap() {
-        const chartProps = this.chart.props
+    @computed get sourceChart(): ChartConfigProps | undefined {
         const sourceChartId = (sourceCharts as any)[
             this.constrainedParams.sourceChartKey
         ]
 
+        return this.props.covidChartAndVariableMeta.charts[sourceChartId]
+    }
+
+    private _updateMap() {
+        const chartProps = this.chart.props
+
         Object.assign(
             chartProps.map,
-            this.props.covidChartAndVariableMeta.charts[sourceChartId]?.map ||
-                this.defaultMapConfig
+            this.sourceChart?.map || this.defaultMapConfig
         )
 
         chartProps.map.targetYear = undefined
         chartProps.map.variableId = this.currentYVarId
     }
 
+    private _updateTableDisplaySettings() {
+        const tableDisplaySettingsByDimensionProperty: {
+            [dimensionProperty: string]:
+                | OwidVariableTableDisplaySettings
+                | undefined
+        } = {}
+
+        // get table display settings for every dimension on source chart
+        this.sourceChart?.dimensions.forEach(
```

Comment:
Ah, so right now we are just extracting color styles from the "sourcecharts". The variables on the "sourceCharts" are not equal to the columns in the Explorer. So the dimensions don't necessarily align. For the Explorer we'd probably want to hard code the tableDisplaySettings into the column specs.

Reply:
:( Done.